desktop_version/src/FileSystemUtils.cpp

```diff
@@ -164,66 +164,132 @@ char *FILESYSTEM_getUserLevelDirectory(void)
 	return levelDir;
 }
 
-bool FILESYSTEM_directoryExists(const char *fname)
+static bool FILESYSTEM_exists(const char *fname)
 {
 	return PHYSFS_exists(fname);
 }
 
 void FILESYSTEM_mount(const char *fname)
 {
-	std::string path(PHYSFS_getRealDir(fname));
-	path += PHYSFS_getDirSeparator();
-	path += fname;
-	if (!PHYSFS_mount(path.c_str(), NULL, 0))
+	const char* real_dir = PHYSFS_getRealDir(fname);
+	const char* dir_separator;
+	char path[MAX_PATH];
+
+	if (real_dir == NULL)
+	{
+		printf(
+			"Could not mount %s: real directory doesn't exist\n",
+			fname
+		);
+		return;
+	}
+
+	dir_separator = PHYSFS_getDirSeparator();
+
+	SDL_snprintf(path, sizeof(path), "%s%s%s", real_dir, dir_separator, fname);
+
+	if (!PHYSFS_mount(path, NULL, 0))
 	{
 		printf("Error mounting: %s\n", PHYSFS_getErrorByCode(PHYSFS_getLastErrorCode()));
 	}
 	else
 	{
-		graphics.assetdir = path.c_str();
+		graphics.assetdir = std::string(path);
 	}
 }
 
 bool FILESYSTEM_assetsmounted = false;
 
 void FILESYSTEM_mountassets(const char* path)
 {
-	const std::string _path(path);
+	const size_t path_size = SDL_strlen(path);
+	char filename[MAX_PATH];
+	char zip_data[MAX_PATH];
+	const char* zip_normal;
+	char dir[MAX_PATH];
+
+	SDL_strlcpy(
+		filename,
+		&path[7],
+		VVV_min((path_size - 14) + 1, sizeof(filename))
```

Comment:
A couple magic numbers that need comments (mostly stemming from me not commenting them originally)...

Reply:
You weren't the person that added this code (unless you're talking about being the person who merged it in without asking for it to be commented). But yeah, I'll comment it.

Reply:
Done.